### PR TITLE
Add basic error handling in the XML parser.

### PR DESCRIPTION
**Pull request by francesco-romano**
Currently the only error returned by the parser is for invalid or not well format XML where the parser itself will fail. For any other "semantic" validation there is no way to return an error to the caller, just prints to stderr.

One option could of course being to return all parsed data and do a validation after but this would implies some changes and keeping track of discarded parsed text.

This change introduces a simple state that is propagated to all parser elements. The state is currently a simple bool, enough to report an error but can be easily extended if needed.

The code has a lot of TODOs where to put error notification. Currently this change reports only an error in case two joints are named the same, and a test has been added to verify it.

### CODE REVIEW

File: src/model_io/xml/include/iDynTree/XMLParser.h [hunk 1]
```diff
@@ -198,12 +200,12 @@ class iDynTree::XMLParser {
      *
      * By specifying a new factory function, it is possible to change how the XML document will be
      * represented in memory.
-     * The signature of the function is `(void) -> std::shared_ptr<XMLDocument>`, i.e. a function
-     * accepting no arguments and returning a `std::shared_ptr` to an `XMLDocument` object.
+     * The signature of the function is `(MLParserState&) -> std::shared_ptr<XMLDocument>`, i.e. a function
```
Comment (traversaro) [inline src/model_io/xml/include/iDynTree/XMLParser.h:203]:
```suggestion
     * The signature of the function is `(XMLParserState&) -> std::shared_ptr<XMLDocument>`, i.e. a function
```

**Comment by traversaro** [comment]
Thanks @francesco-romano ! Please:
* Minor version to 8.999.0, as this is an API breaking change, so next release will be iDynTree 9, and bumping the minor number to such an high number is a good way or remembering to do it.
* Add line to the CHANGELOG.md

File: src/model_io/xml/include/iDynTree/XMLParser.h (hunk 1 above)
Comment (francesco-romano) [reply src/model_io/xml/include/iDynTree/XMLParser.h:203]: Thanks

**Comment by francesco-romano** [comment]
Done

**Comment by traversaro** [comment]
Thanks!

**Pull request merged**